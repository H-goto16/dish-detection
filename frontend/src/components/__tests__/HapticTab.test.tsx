import renderer from 'react-test-renderer';
import { HapticTab } from '../HapticTab';

// Mock expo-haptics
jest.mock('expo-haptics', () => ({
  impactAsync: jest.fn(),
  ImpactFeedbackStyle: {
    Light: 'light',
    Medium: 'medium',
    Heavy: 'heavy',
  },
}));

// Mock @react-navigation/elements
jest.mock('@react-navigation/elements', () => ({
  PlatformPressable: ({
    children,
    ...props
  }: { children?: React.ReactNode; [key: string]: unknown }) => {
    const { View } = require('react-native');
    return <View {...props}>{children}</View>;
  },
}));

describe('HapticTab', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    process.env.EXPO_OS = 'ios';
  });

  const mockProps = {
    onPress: jest.fn(),
    onPressIn: jest.fn(),
    children: 'Tab Content',
  };

  it('renders correctly', () => {
    const tree = renderer.create(<HapticTab {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const component = renderer.create(<HapticTab {...mockProps} />);
    expect(component).toBeTruthy();
  });

  it('renders with different props', () => {
    const customProps = {
      onPress: jest.fn(),
      children: 'Custom Tab',
      testID: 'custom-tab',
    };

    const tree = renderer.create(<HapticTab {...customProps} />).toJSON();
    expect(tree).toBeTruthy();
  });

  it('handles empty children', () => {
    const propsWithEmptyChildren = {
      onPress: jest.fn(),
      children: null,
    };

    const component = renderer.create(
      <HapticTab {...propsWithEmptyChildren} />,
    );
    expect(component.toJSON()).toBeTruthy();
  });
});
