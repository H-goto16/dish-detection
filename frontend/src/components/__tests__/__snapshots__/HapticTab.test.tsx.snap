// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HapticTab renders correctly 1`] = `
<View
  onPress={[MockFunction]}
  onPressIn={[Function]}
>
  Tab Content
</View>
`;
